src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/Extensions/StringExtensions.cs

```diff
@@ -105,6 +105,65 @@ public static string RemoveBlankLines(this string value)
                 .Trim();
         }
 
+        /// <summary>
+        /// Removes the duplicate string from the parameter name to work around the roslyn issue.
+        /// https://github.com/dotnet/roslyn/issues/26292.
+        /// </summary>
+        /// <remarks>
+        /// If a character is not allowed in an identifier name in .NET, roslyn duplicates the string following the
+        /// first occurence of that special character in the complied xml.
+        /// e.g if the param name is service-Catalog-Id-1, roslyn will convert it to
+        /// service-Catalog-Id-1-Catalog-Id-1 and this method will remove the duplicate string -Catalog-Id-1 and
+        /// return service-Catalog-Id-1
+        /// Also @ is allowed at the start of the identifier name, so roslyn duplicates the string following the
+        /// second occurence of that special character in the complied xml.
+        /// e.g if the param name is @skip@skip, roslyn will convert it to
+        /// @skip@skip@skip and this method will remove the duplicate string @skip and
+        /// return @skip@skip
+        /// </remarks>
+        /// <param name="value">The param name to update.</param>
+        /// <returns>The updated value.</returns>
+        public static string RemoveRoslynDuplicateString(this string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return value;
+            }
+
+            // Only alphanumeric or hyphen characters are allowed in an identifier name.
+            // So look for special character other than these.
+            string firstOccuredSpecialCharacter = Regex.Match(value, "[^a-zA-Z0-9_]").Value;
+
+            if (string.IsNullOrWhiteSpace(firstOccuredSpecialCharacter))
+            {
+                return value;
+            }
+
+            var specialCharIndex = value.IndexOf(firstOccuredSpecialCharacter.ToCharArray()[0]);
+
+            // @ is allowed at the start of the identifier name, so look for second occurence of it.
+            // e.g. if a param name is @skip@skip, roslyn will compile it as @skip@skip@skip.
+            if (firstOccuredSpecialCharacter == "@")
+            {
+                specialCharIndex = value.IndexOf(firstOccuredSpecialCharacter.ToCharArray()[0], 1);
+            }
+
+            if (specialCharIndex == -1)
+            {
+                return value;
+            }
+
+            // Divide the string after special character into two halves and check if they are equal.
+            // If equal take only first half of string. if not equal return value as is.
+            var valueLengthAfterSpecialChar = value.Length - specialCharIndex;
+            var firstHalf = value.Substring(specialCharIndex, valueLengthAfterSpecialChar / 2);
+            var secondHalf = value.Substring(specialCharIndex + valueLengthAfterSpecialChar / 2,
+                valueLengthAfterSpecialChar / 2);
```

Comment:
I believe this comparison will fail for this case:

a-b-bc

since it is comparing index 1-2 `-b` with 3-4 `-b` and ignoring the last character.

You can first check whether valueLengthAfterSpecialChar is an even number. If it's not, then it's an immediate non-match and we don't need to construct compare first and second half.